docs/markdown/Machine-files.md

```diff
@@ -5,24 +5,52 @@ documentation on the common values used by both, for the specific values of
 one or the other see the [cross compilation](Cross-compilation.md) and [native
 environments](Native-environments.md).
 
+## Data Types
+
+There are three basic data types in a machine file:
+- strings
+- arrays
+- booleans
+
+A string is specified single quoted:
+```ini
+[section]
+option = 'value'
+```
+
+An array is enclosed in square brackets, and must consist of strings or booleans
```

Comment:
Do we also support implicitly defining an array with a single element using string syntax like we do for the `[binaries]` section? I have a slight preference for not doing that if we do, but if we decide to support that it should be documented.

Reply:
It's not done in the machine file, but probably a stringlistify() somewhere when it gets the value of binaries. Array options works like that too, `c_args='-DFOO'` works.